src/libraries/System.Net.Ping/tests/FunctionalTests/PingTest.cs

```diff
@@ -252,57 +253,47 @@ public async Task SendPingAsyncWithIPAddressAndTimeoutAndBuffer()
                 });
         }
 
-        [PlatformSpecific(TestPlatforms.AnyUnix)]  // On Unix, Non-root pings cannot send arbitrary data in the buffer, and do not receive it back in the PingReply.
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
         [Fact]
         public void SendPingWithIPAddressAndTimeoutAndBuffer_Unix()
         {
-            byte[] buffer = TestSettings.PayloadAsBytes;
             IPAddress localIpAddress = TestSettings.GetLocalIPAddress();
 
+            // On Unix, Non-root pings cannot send arbitrary data in the buffer, and do not receive it back in the PingReply.
+            byte[] buffer = Capability.CanUseRawSockets(localIpAddress.AddressFamily) || PlatformDetection.IsOSXLike
```

Comment:
Seems to repeat a lot. We may consider helper function to simply give buffer back.